### ISSUE TITLE
Lab 09-5 Sigmoid used instead of Softmax 

### ISSUE THREAD

**Issue by kkweon** [hunkim/DeepLearningZeroToAll#51]
# Problem
In [lab-09-5-softmax_back_prop.py](https://github.com/hunkim/DeepLearningZeroToAll/blob/master/lab-09-5-softmax_back_prop.py),

Although softmax is a generalization of the logistics function, two have different formulas and different derivatives. The current implementation is technically doing **multiple of single logistics regressions** instead of **multi-class classification**. 

What it means *the sum of each output row will not be equal to 1* when using sigmoid instead of softmax

![image](https://cloud.githubusercontent.com/assets/2981167/23538989/bc01df52-ff8c-11e6-85ac-2e451294637c.png)

The sigmoid implementation in the file
```python
def sigma(x):
    #  sigmoid function
    return tf.div(tf.constant(1.0),
                  tf.add(tf.constant(1.0), tf.exp(-x)))


def sigma_prime(x):
    # derivative of the sigmoid function
    return sigma(x) * (1 - sigma(x))
```

# Suggestion
There are two options
1. modify the filename to sigmoid backprop 
2. actually implement correct softmax

First suggestion is preferred since the derivative of softmax can be difficult for beginners and yet it should be clear that `tf.nn.sofmax` is not sigmoid. 

In other files containing softmax in its filename, it's all using `tf.nn.softmax` since it's using tensorflow.

**Comment by hunkim** [comment]
Thanks for your suggestion. I also like the first option.

Cheers,
Sung
